### PR TITLE
Switch to eslint

### PR DESCRIPTION
**Pull request by morrigan**
- Use eslint instead of tslint
 - Use strict mode
 - Fix errors and add lots of typings

Some other smaller things also: 

 - Fix docker production node version to match development's
 - Moved nodemon to be dev dependency
 - Replace npm commands with yarn
 - Try to fix github security alert :]

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -0,0 +1,25 @@
+{
+  "env": {
+    "mocha": true,
+    "node": true,
+    "es6": true
```
Comment (mpetrunic) [inline .eslintrc:5]: typescript? lol

File: .eslintrc [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  "plugins": [
+    "@typescript-eslint"
+  ],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended"
+  ],
+  "rules": {
+    "@typescript-eslint/no-require-imports": "error",
+    "@typescript-eslint/ban-ts-ignore": "off",
```
Comment (mpetrunic) [inline .eslintrc]: ? at least warn

File: .eslintrc [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    "@typescript-eslint/no-unused-vars": ["warn", {
```
Comment (mpetrunic) [inline .eslintrc]: Can be error

File: package.json [hunk 4]
```diff
@@ -10,33 +10,32 @@
...
+    "prebuild": "yarn run clean && yarn run init",
     "build": "tsc --incremental --sourceMap --outDir dist",
     "compile": "tsc",
-    "pretest": "npm run compile && npm run lint",
+    "pretest": "yarn run compile && yarn run lint",
     "test:unit": "NODE_ENV=test mocha -r dotenv/config -r ts-node/register \"test/unit/**/*.ts\"",
     "test": "yarn run test:unit & yarn run test:e2e",
     "test:e2e": "SERVER_PORT=0 mocha --timeout 30000 -r ts-node/register -r dotenv/config \"test/e2e/**/*.ts\"",
-    "coverage": "nyc --reporter=text --all --include \"src/**/*.js\" npm test",
+    "coverage": "nyc --reporter=text --all --include \"src/**/*.js\" yarn test",
```
Comment (mpetrunic) [inline package.json]: test:unit

File: src/App.ts [hunk 5]
```diff
@@ -1,28 +1,30 @@
...
+    private helpController?: HelpController;
 
     constructor() {
         this.express = express();
         // add before route middleware's here
-        this.express.use(morgan("short", { stream: morganLogger }));
+        // @ts-ignore-next-line
+        this.express.use(morgan("short", {stream: morganLogger}));
         this.express.use(bodyParser.json());
+        // @ts-ignore-next-line
```
Comment (mpetrunic) [inline src/App.ts]: ? why this

**Review by mpetrunic — CHANGES_REQUESTED** [review]
```
"@typescript-eslint/explicit-function-return-type": "error",
 "@typescript-eslint/explicit-member-accessibility": "error"
"@typescript-eslint/no-explicit-any": "error"
"no-console": "error"
```

File: .eslintrc (hunk 1 above)
Comment (morrigan) [reply .eslintrc:5]: this might have been c/p from CG :]

File: src/App.ts (hunk 5 above)
Comment (morrigan) [reply src/App.ts]: Types going crazy, couldn't figure it out.

File: .eslintrc (hunk 1 above)
Comment (morrigan) [reply .eslintrc:5]: Also, not sure if you are referring to es6 with the comment but that's required if using es6 global variables

File: src/App.ts (hunk 5 above)
Comment (morrigan) [reply src/App.ts]: I guess I could remove helmet types and declare that module

Comment (mpetrunic) [reply src/App.ts]: Enable esModuleInterop flag in tsconfig and it should be fine

Comment (morrigan) [reply src/App.ts]: that has been already enabled

Comment (mpetrunic) [reply src/App.ts]: try import * as helmet from "helmet"

Comment (morrigan) [reply src/App.ts]: nop, but hacked something out by lowering version and explicitly converting ¯\_(ツ)_/¯

File: package.json [hunk 6]
```diff
@@ -29,11 +29,13 @@
   },
   "dependencies": {
     "@hapi/joi": "^15.1.0",
+    "@types/helmet": "^0.0.45",
+    "@types/morgan": "^1.7.37",
```
Comment (mpetrunic) [inline package.json]: we aren't suppose to package types into production build unless it's a library that exposes that type

File: src/App.ts [hunk 7]
```diff
@@ -21,11 +21,9 @@ export class App implements Service {
     constructor() {
         this.express = express();
         // add before route middleware's here
-        // @ts-ignore-next-line
-        this.express.use(morgan("short", {stream: morganLogger}));
+        this.express.use(morgan("short", {stream: morganLogger}) as RequestHandler);
         this.express.use(bodyParser.json());
-        // @ts-ignore-next-line
-        this.express.use(helmet());
+        this.express.use(helmet() as RequestHandler);
```
Comment (mpetrunic) [inline src/App.ts:26]: https://stackoverflow.com/questions/56116660/how-to-use-helmet-js-from-typescript

Comment (mpetrunic) [reply src/App.ts:26]: google morgan as well, similar solution

File: package.json (hunk 6 above)
Comment (morrigan) [reply package.json]: argh, mistakenly went into prod depenencies by constant removing and adding

File: src/App.ts (hunk 7 above)
Comment (morrigan) [reply src/App.ts:26]: I already did that but that doesn't work. If you want to try, feel free

**Pull request merged**